github-pullrequest-plugin/src/main/java/org/jenkinsci/plugins/github/pullrequest/events/impl/GitHubPRSkipUnmmatchNumber.java

```diff
@@ -42,11 +42,11 @@ public GitHubPRCause check(GitHubPRTrigger gitHubPRTrigger, GHPullRequest remote
         }
 
         if (remotePR != null && remotePR.getNumber() == getNumber()) {
-            return new GitHubPRCause(remotePR, "Skipping matched pr " + remotePR.getNumber(), true);
+            return new GitHubPRCause(remotePR, "Building matched pr " + remotePR.getNumber(), false);
```

Comment:
That will trigger build, you must reverse logic and not skip. skip== true => skip checking other events.